lib/scholar/impute/simple_imputer.ex

```diff
@@ -0,0 +1,242 @@
+
+    input_rank = Nx.rank(x)
+
+    if input_rank != 2 do
+      raise ArgumentError, "Wrong input rank. Expected: 2, got: #{inspect(input_rank)}"
+    end
+
+    if validated_opts[:missing_values] != :nan and
+         Nx.any(Nx.is_nan(x)) == Nx.tensor(1, type: :u8) do
+      raise ArgumentError,
+            ":missing_values other than :nan possible only if there is no Nx.Constant.nan() in the array"
+    end
+
+    {type, _num_bits} = Nx.type(x)
+
+    x =
+      if validated_opts[:strategy] == :constant do
+        cond do
+          is_float(validated_opts[:fill_value]) and type in [:s, :u] ->
+            Nx.as_type(x, :f32)
+
+          is_integer(validated_opts[:fill_value]) and type in [:f, :bf] ->
+            {fill_value_type, _} = Nx.type(validated_opts[:fill_value])
+
+            raise ArgumentError,
+                  "Wrong type of `:fill_value` for the given data. Expected: :f or :bf, got: #{inspect(fill_value_type)}"
+
+          true ->
+            x
+        end
+      else
+        x
+      end
+
+    x_type = Nx.type(x)
+
+    x =
+      if validated_opts[:missing_values] != :nan,
+        do: Nx.select(Nx.equal(x, validated_opts[:missing_values]), Nx.Constants.nan(), x),
+        else: x
+
+    {_num_rows, num_cols} = Nx.shape(x)
+
+    statistics =
+      cond do
+        opts[:strategy] == :mean ->
+          mean_op(x)
+
+        opts[:strategy] == :median ->
+          median_op(x)
+
+        opts[:strategy] == :mode ->
+          mode_op(x, type: x_type)
+
+        true ->
+          Nx.broadcast(validated_opts[:fill_value], {num_cols})
+      end
+
+    missing_values = validated_opts[:missing_values]
+    %__MODULE__{statistics: statistics, missing_values: missing_values}
+  end
+
+  defnp mean_op(x) do
+    mask = not Nx.is_nan(x)
+    denominator = Nx.sum(mask, axes: [0])
+    temp = Nx.select(mask, x, 0)
+    nominator = Nx.sum(temp, axes: [0])
+
+    Nx.select(
+      denominator != 0,
+      nominator / denominator,
+      Nx.Constants.nan()
+    )
+  end
+
+  defnp median_op(x) do
+    {num_rows, num_cols} = Nx.shape(x)
+    x = Nx.sort(x, axis: 0)
+    tensor = Nx.iota(x, axis: 0) * num_rows
+    res = Nx.argsort(tensor, axis: 0)
+    x = Nx.take_along_axis(x, res, axis: 0)
+
+    indices = Nx.broadcast(num_rows - 1, {num_cols})
+    indices = indices - Nx.sum(Nx.is_nan(x), axes: [0])
+    half_indices = indices / 2
+    floor = Nx.as_type(Nx.floor(half_indices), :s64) |> Nx.new_axis(0)
+    ceil = Nx.as_type(Nx.ceil(half_indices), :s64) |> Nx.new_axis(0)
+    nums1 = Nx.take_along_axis(x, floor, axis: 0)
+    nums2 = Nx.take_along_axis(x, ceil, axis: 0)
+    ((nums1 + nums2) / 2) |> Nx.squeeze()
+  end
+
+  defnp mode_op(x, opts \\ []) do
+    type = opts[:type]
+
+    axis = 0
+
+    axis_length = Nx.axis_size(x, axis)
+    x = Nx.sort(x, axis: 0)
+    tensor = Nx.multiply(Nx.iota(x, axis: axis), axis_length)
+    res = Nx.argsort(tensor, axis: axis)
```

Comment:
Same issue here, this seems to be the iota itself

Reply:
Same here